AG-8557 Change double click to not reset zoom if over a node

lsjroberts opened the pull request:
https://ag-grid.atlassian.net/browse/AG-8301
https://ag-grid.atlassian.net/browse/AG-8557

Adds a `nodeDatumManager` to the module context. This allows zoom to check if double clicking over a node and then prevent a reset of the zoom.

It will also be useful for the context menu to give options based on the clicked datum later.

Questions:
- I'm very open to name suggestions, like something with the words of `node`, `datum`, `event` manager?
- There's a `chartEventsManager` that I was originally going to use for this, but I'm not sure is a great fit given the tracked state (which will likely be expanded upon by other features). Which also begs the question if the chart events manager can be improved upon and not just a random collection of things.

lsjroberts committed:
AG-8557 Add nodeDatumManager

lsjroberts committed:
AG-8557 Change double click to not reset zoom if over a node

alantreadway commented:
This seems to overlap a bit conceptually with `HighlightManager` - should we merge the two?

lsjroberts committed:
AG-8557 Replace nodeDatumManager with highlightManager

lsjroberts committed:
AG-8557 Add double click reset on zoom by specific axis

lsjroberts commented:
I've updated it to use the `highlightManager`, I haven't renamed that yet. I'll wait to see what happens with context menu to see if that inspires a better name for any extra functionality.

Also added resetting on a specific axis when clicked.

alantreadway reviewed (approved):
LGTM
